ecogvis/functions/nwb_copy_file.py

```diff
@@ -381,3 +458,52 @@ def copy_obj(obj_old, nwb_old, nwb_new):
             power=obj_old.power,
             electrodes=elecs_region
         )
+
+    # Survey tables ------------------------------------------------------------
+    if obj_old.neurodata_type == 'SurveyTable':
+        if obj_old.name == 'nrs_survey_table':
+            n_rows = len(obj_old['nrs_pain_intensity_rating'].data[:])
+            for i in range(n_rows):
+                nrs_survey_table.add_row(
+                    nrs_pain_intensity_rating=obj_old['nrs_pain_intensity_rating'].data[i],
+                    nrs_relative_pain_intensity_rating=obj_old['nrs_relative_pain_intensity_rating'].data[i],
+                    nrs_pain_unpleasantness=obj_old['nrs_pain_unpleasantness'].data[i],
+                    nrs_pain_relief_rating=obj_old['nrs_pain_relief_rating'].data[i],
+                    unix_timestamp=obj_old['unix_timestamp'].data[i]
+                )
+            return nrs_survey_table
+
+        elif obj_old.name == 'vas_survey_table':
+            n_rows = len(obj_old['vas_pain_intensity_rating'].data[:])
+            for i in range(n_rows):
+                vas_survey_table.add_row(
+                    vas_pain_intensity_rating=obj_old['vas_pain_intensity_rating'].data[i],
+                    vas_pain_relief_rating=obj_old['vas_pain_relief_rating'].data[i],
+                    vas_relative_pain_intensity_rating=obj_old['vas_relative_pain_intensity_rating'].data[i],
+                    vas_pain_unpleasantness=obj_old['vas_pain_unpleasantness'].data[i],
+                    unix_timestamp=obj_old['unix_timestamp'].data[i],
+                )
+            return vas_survey_table
+
+        elif obj_old.name == 'mpq_survey_table':
+            n_rows = len(obj_old['throbbing'].data[:])
+            for i in range(n_rows):
+                mpq_survey_table.add_row(
```

Comment:
You should be able to do something much more compact for these. Something like:

`mpq_survey_table.add_row({key: val.data[i] for key, val in obj_old.items()})`

Reply:
solved with 9e76d92af879526cdc0b8c64e46b98857f903f2b